hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/MerkleTestBase.java

```diff
@@ -290,12 +331,24 @@ protected byte[] writeTree(@NonNull final MerkleNode tree, @NonNull final Path t
         return byteOutputStream.toByteArray();
     }
 
-    /** A convenience method used to deserialize a merkle tree */
+    /**
+     * A convenience method used to deserialize a merkle tree
+     */
     protected <T extends MerkleNode> T parseTree(@NonNull final byte[] state, @NonNull final Path tempDir)
             throws IOException {
         final var byteInputStream = new ByteArrayInputStream(state);
         try (final var in = new MerkleDataInputStream(byteInputStream)) {
             return in.readMerkleTree(tempDir, 100);
         }
     }
+
+    @AfterEach
+    void cleanUp() throws IOException, NoSuchFieldException, IllegalAccessException {
```

Comment:
Is it really needed? MerkleDb storage dirs are created in the temp folder, like `swirlds-tmp`, which is expected to be deleted before each test run.

Reply:
Yes, it's really needed because MerkleDB instances are cached in the static context. And even if we use different directory, it doesn't help much when we create a DB with the same name. It fails with an exception